spec/unit/query_azure_mock.rb

```diff
@@ -7,10 +8,31 @@ def setup_query_azure_mock
     stub_query_azure (@connection)
   end
 
+  def mock_list_images
+    create_service
+    stub_list_images(@service)
+  end
+
   def create_connection
     @connection = Azure::Connection.new(TEST_PARAMS)
   end
 
+  def create_service
+    @service = Azure::ServiceManagement::ASMInterface.new(TEST_PARAMS)
+  end
+
+  def stub_list_images(service)
+    stub_query_azure(service.connection)
+  end
+
+  def stub_delete_server(service)
+    stub_query_azure(service.connection)
+  end
+
+  def stub_show_server(service)
+    stub_query_azure(service.connection)
+  end
```

Comment:
@NimishaS I am using different methods to understand the code flow for e.g. in azure_list_image we call service.list_image and I want anyone to understand I am stubbing the list_image method and stub_query_azure is just a method call and not the repetition of the code.